### PR TITLE
Restore hashtag suggestions from local tag history

### PR DESCRIPTION
**Pull request by ClearlyClaire**
I'm not happy about mixing `ImmutableJS` and native lists/objects here, but it seems like the best way to make it work both for when the REST API hasn't replied yet and for results are gotten from it…

~~EDIT: also not super happy about it claiming there has been 0 uses in the week when the tag data isn't from API results~~

Example when `#CatPosting` isn't approved (but has been used locally) but `#cat` is:
Peek 21-08-2019 13-13

### CODE REVIEW

File: app/javascript/mastodon/reducers/compose.js [hunk 1]
```diff
@@ -210,7 +210,7 @@ const mergeLocalHashtagResults = (suggestions, prefix, tagHistory) => {
   prefix = prefix.toLowerCase();
   if (suggestions.length < 4) {
     const localTags = tagHistory.filter(tag => tag.toLowerCase().startsWith(prefix) && !suggestions.some(suggestion => suggestion.type === 'hashtag' && suggestion.name.toLowerCase() === tag.toLowerCase()));
-    return localTags.slice(0, 4 - suggestions.length).toJS().map(tag => ({ type: 'hashtag', name: tag, history: [] })).concat(suggestions);
+    return suggestions.concat(slice(0, 4 - suggestions.length).toJS().map(tag => ({ type: 'hashtag', name: tag })));
```
Comment (ykzts) [inline app/javascript/mastodon/reducers/compose.js]: `slice` ?

Comment (ThibG) [reply app/javascript/mastodon/reducers/compose.js]: Sorry, messed that one up ^^'
fixed

Comment (Gargron) [inline app/javascript/mastodon/reducers/compose.js]: Hold up what is it slicing now?

**Review by ykzts — APPROVED** [review]

**Comment by Gargron** [comment]
> ~~EDIT: also not super happy about it claiming there has been 0 uses in the week when the tag data isn't from API results~~

It should be possible to fetch the data for those tags

**Pull request merged**